include/boost/windows_sspi/detail/context_impl.hpp

```diff
@@ -11,150 +11,43 @@
 #ifndef BOOST_WINDOWS_SSPI_DETAIL_CONTEXT_IMPL_HPP
 #define BOOST_WINDOWS_SSPI_DETAIL_CONTEXT_IMPL_HPP
 
-#include <boost/windows_sspi/error.hpp>
-
-#include <boost/windows_sspi/detail/sspi_functions.hpp>
 #include <boost/windows_sspi/detail/config.hpp>
+#include <boost/windows_sspi/detail/cryptographic_provider.hpp>
+#include <boost/windows_sspi/detail/sspi_functions.hpp>
+#include <boost/windows_sspi/detail/win32_crypto.hpp>
+#include <boost/windows_sspi/detail/win32_file.hpp>
+#include <boost/windows_sspi/error.hpp>
 
-#include <boost/winapi/file_management.hpp>
-#include <boost/winapi/handles.hpp>
-#include <boost/winapi/access_rights.hpp>
-
-// TODO: Consider adding function prototypes to avoid this include
-#include <wincrypt.h>
+#include <boost/assert.hpp>
 
 namespace boost {
 namespace windows_sspi {
 namespace detail {
 
-struct cert_chain_context {
-  ~cert_chain_context() {
-    CertFreeCertificateChain(ptr);
-  }
-
-  PCCERT_CHAIN_CONTEXT ptr = nullptr;
-};
-
-struct cert_chain_engine {
-  ~cert_chain_engine() {
-    CertFreeCertificateChainEngine(ptr);
-  }
-
-  HCERTCHAINENGINE ptr = nullptr;
-};
-
 struct context_impl {
   context_impl()
     : m_cert_store(CertOpenStore(CERT_STORE_PROV_MEMORY, 0, 0, 0, nullptr)) {
     if (m_cert_store == nullptr) {
       throw boost::system::system_error(error::make_error_code(boost::winapi::GetLastError()), "CertOpenStore");
```

Comment:
use `BOOST_THROW_EXCEPTION` instead of raw `throw`

Reply:
OK. Will do.